src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/SourceBuiltArtifactsTests.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+public class SourceBuiltArtifactsTests : SmokeTests
+{
+    public SourceBuiltArtifactsTests(ITestOutputHelper outputHelper) : base(outputHelper) { }
+
+    [SkippableFact(Config.RunningInCIEnv, skipOnNullOrWhiteSpace: true, skipOnFalse: true)]
```

Comment:
I want to leave the door open for other SB artifact tests to exist. Those would likely be tests that should be run in both CI and dev scenarios. That isn't the case for this version file test. So making the tests conditional on the presence of `SMOKE_TESTS_SOURCEBUILT_ARTIFACTS_PATH` isn't enough. We also need a marker that only gets set in the case of CI. To me, it didn't make sense to introduce a variable that's specific to testing version file. That's scoped much too narrowly. That's why I thought it made sense to use a more general variable that indicated whether it's being run from CI. That generality would make it applicable for other tests to use in the future that have a similar requirement.

Reply:
What makes this test special that you wouldn't want to run it in a local run?  You have all of the information present to validate it don't you?  In general it would be preferred to have as few conditional tests as possible.

Reply:
It tests that the commit SHA is not a bunch of 0s which would be the default value if Arcade didn't have the pipeline's variable. This comes back to this: https://github.com/dotnet/installer/pull/15353#discussion_r1089208776

Reply:
Yes, but to be safe I think we'd want to handle a scenario of a dev environment where someone may have a variable set that caused Arcade to initial `RepositoryCommit` to a real value, not all zeroes. So if `RunningInCIEnv` is false, I think it would just skip the check that it's not all zeroes. But the other validation of length and character type would still be done.